Source/Particles/Deposition/ChargeDeposition.H

```diff
@@ -201,6 +201,322 @@ void doChargeDepositionShapeN (const GetParticlePosition& GetPosition,
 #endif
         }
         );
+#if defined(WARPX_USE_GPUCLOCK)
+    if( load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::GpuClock) {
+        amrex::Gpu::streamSynchronize();
+        *cost += *cost_real;
+        amrex::The_Managed_Arena()->free(cost_real);
+    }
+#endif
+
+#ifndef WARPX_DIM_RZ
+    amrex::ignore_unused(n_rz_azimuthal_modes);
+#endif
+}
+
+#if defined(AMREX_USE_CUDA) || defined(AMREX_USE_HIP)
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void addLocalToGlobal (const amrex::Box& bx,
+                       const amrex::Array4<amrex::Real>& global,
+                       const amrex::Array4<amrex::Real>& local) noexcept
+{
+    const auto lo  = amrex::lbound(bx);
+    const auto len = amrex::length(bx);
+    for (int icell = threadIdx.x; icell < bx.numPts(); icell += blockDim.x)
+    {
+        int k =  icell / (len.x*len.y);
+        int j = (icell - k*(len.x*len.y)) /   len.x;
+        int i = (icell - k*(len.x*len.y)) - j*len.x;
+        i += lo.x;
+        j += lo.y;
+        k += lo.z;
+        amrex::Gpu::Atomic::AddNoRet( &global(i, j, k), local(i, j, k));
+    }
+}
+#endif
+
+/* \brief Perform charge deposition on a tile using shared memory
```

Comment:
The term tile is maybe confusing here, no?

Reply:
This is exactly the same usage as in the existing `doChargeDepositionShapeN`

Reply:
Oh, heh, this changes for that, too. I blame @atmyers